packages/wrangler/src/inspect.ts

```diff
@@ -148,13 +148,19 @@ export default function useInspector(props: InspectorProps) {
    * of the component lifecycle. Convenient.
    */
   useEffect(() => {
-    serverRef.current.listen(props.port);
-    return () => {
-      serverRef.current.close();
-      // Also disconnect any open websockets/devtools connections
-      wsServerRef.current.clients.forEach((ws) => ws.close());
-      wsServerRef.current.close();
-    };
+    if (serverRef.current) {
```

Comment:
I think we should assert here. It's a big problem if it's not defined, so we should probably throw.

Reply:
An alternative is that we capture these values in a const outside the callback scope...

```ts
      const serverInst = serverRef.current;
      const wsServerInst = wsServerRef.current;
      return () => {
        serverInst.close();
        // Also disconnect any open websockets/devtools connections
        wsServerInst.clients.forEach((ws) => ws.close());
        wsServerInst.close();
      };
```

But I wasn't sure if we could be confident that this would never change between calls to `useEffect()` and its cleanup function...